test/specs/api_designer/oas_api_endpoints_test.js

```diff
@@ -0,0 +1,174 @@
+    let searchCriteria = 'ip';
+    let expectedEndpointList = [
+      {
+        endpoint: '/ip',
+        method: 'GET'
+      },
+      {
+        endpoint: '/ips',
+        method: 'GET'
+      },
+      {
+        endpoint: '/ip',
+        method: 'POST'
+      }
+    ];
+    endpoints_page.searchEndpointsByName(searchCriteria);
+    createdEndpointsList.forEach (e => {
+      verifyDisplayedEndpoints(e, expectedEndpointList);
+    });
+  });
+
+  it('User can filter endpoints by method', () => {
+    let methodFilter = 'GET';
+    let expectedEndpointList = [
+      {
+        endpoint: '/ip',
+        method: 'GET'
+      },
+      {
+        endpoint: '/ips',
+        method: 'GET'
+      },
+      {
+        endpoint: '/headers',
+        method: 'GET'
+      }
+    ];
+    endpoints_page.clearSerachCriteria();
+    endpoints_page.filterEndpointsByMethod(methodFilter);
+    createdEndpointsList.forEach (e => {
+      verifyDisplayedEndpoints(e, expectedEndpointList);
+    });
+  });
+
+  it('User can filter endpoints by endpoint name and method', () => {
+    let methodFilter = 'POST';
+    let searchCriteria = 'header';
+    let expectedEndpointList = [
+      {
+        endpoint: '/headers',
+        method: 'POST'
+      }
+    ];
+    endpoints_page.searchEndpointsByName(searchCriteria);
+    endpoints_page.filterEndpointsByMethod(methodFilter);
+    createdEndpointsList.forEach (e => {
+      verifyDisplayedEndpoints(e, expectedEndpointList);
+    });
+  });
+
+  it('User can modify endpoints and save API', () => {
+    browser.refresh();
+    let oldEendpointSelector = $('//a[contains(@href, "-ipGET")]');
+    let newEendpointSelector = $('//a[contains(@href, "-headersHEAD")]');
+    apis_page.EDIT_BUTTON.click();
+    endpoints_page.modifyEndpoint("/ip", "GET", "/headers", "HEAD");
+    apis_page.OAS_SAVE_BUTTON.click();
+    expect(apis_page.isApiUpdatedPopUpDisplayed()).to.be.true; 
+    browser.refresh();
+    endpoints_page.OAS_ENDPOINTS_BUTTON.click();
+    wdioExpect(oldEendpointSelector).not.toBeDisplayed();
+    wdioExpect(newEendpointSelector).toBeDisplayed();
+  });
+
+  it('User can remove endpoints and save API', () => {
+    browser.refresh();
+    let removeEendpointSelector = $('//a[contains(@href, "-ipPOST")]');
+    apis_page.EDIT_BUTTON.click();
+    endpoints_page.removeEndpoint("/ip", "POST");
+    apis_page.OAS_SAVE_BUTTON.click();
+    expect(apis_page.isApiUpdatedPopUpDisplayed()).to.be.true; 
+    browser.refresh();
+    endpoints_page.OAS_ENDPOINTS_BUTTON.click();
+    wdioExpect(removeEendpointSelector).not.toBeDisplayed();
+  });
+
+  function verifyDisplayedEndpoints(endpointObject, expectedEndpoints) {
+    let endpointUrlLink = encodeURIComponent(endpointObject.endpoint.split('/').join('-')) + endpointObject.method;
+    let endpointSelector = $(`//a[contains(@href, "${endpointUrlLink}")]`);
+    if (JSON.stringify(expectedEndpoints).includes(JSON.stringify(endpointObject))){
```

Comment:
I think that this is a place where comment (with brief explanation) will help a lot in the future:)